src/pipeline.cpp

```diff
@@ -351,7 +350,12 @@ static void buildEarlySimplificationPipeline(ModulePassManager &MPM, PassBuilder
         FPM.addPass(DCEPass());
         FPM.addPass(SimplifyCFGPass(basicSimplifyCFGOptions()));
         if (O.getSpeedupLevel() >= 1) {
+#if JL_LLVM_VERSION >= 160000
+            // TODO check the LLVM 15 default.
+            FPM.addPass(SROAPass(SROAOptions::PreserveCFG));
```

Comment:
The current default is SROAOptions::ModifyCFG, the old default was PreserveCFG

https://github.com/llvm/llvm-project/blob/97b989b27496d5aeadb3b90cbb9305ddcd9e35d2/llvm/include/llvm/Transforms/Scalar.h#L66

https://github.com/llvm/llvm-project/blob/97b989b27496d5aeadb3b90cbb9305ddcd9e35d2/llvm/lib/Passes/PassBuilder.cpp#L1023-L1024

ModifyCFG was added in https://github.com/llvm/llvm-project/commit/4f7e5d22060e8a89237ffb93c3e7be6e92fee8fe#diff-2b09f41dec5c9d1fbe844989c38f6bb7db588799b97b4bed7ad173ce7f348280R1119 for llvm-16

Reply:
Thanks for digging that up. So we should use PreserveCFG late in the pipeline and ModifyCFG early on.